VirtoCommerce.CustomerModule.Web/Controllers/Api/CustomerModuleController.cs

```diff
@@ -143,6 +144,29 @@ public IHttpActionResult DeleteMembers([FromUri] string[] ids)
             return StatusCode(HttpStatusCode.NoContent);
         }
 
+        /// <summary>
+        /// Delete planty members
+        /// </summary>
+        /// <remarks>Delete planty members by search criteria of members.</remarks>
+        /// <param name="criteria">concrete instance of SearchCriteria type type will be created by using PolymorphicMemberSearchCriteriaJsonConverter</param>
+        [HttpPost]
+        [Route("members/bulk-delete")]
+        [ResponseType(typeof(void))]
+        [CheckPermission(Permission = CustomerPredefinedPermissions.Delete)]
+        public IHttpActionResult BulkDeleteMembersBySearchCriteria(MembersSearchCriteria criteria)
+        {
+            int currentTake = criteria.Take;
+            criteria.Take = int.MaxValue;
+            var result = _memberSearchService.SearchMembers(criteria);
```

Comment:
Use int.Max value for loading all members may cause serious problems with performance and machine resources usage.
Need to wrap these methods (Search and Delete) in the loop which will execute these methods from each page.
Also, if possible, I would try to avoid using the request for getting the total number of members.